### PR TITLE
Pd partner assessment written assessment and drop down

### PR DESCRIPTION
**Pull request by strempe**
## What's this PR do?
Creates a text box for a written assessment, drop down for selecting a 1 through 5 rating, save button to save Outreach Effectiveness Assessment.

## Screenshots (if appropriate)
screen shot 2017-07-25 at 12 10 06 pm

## Any background context you want to provide?
There is no added CSS.

### CODE REVIEW

File: gulp/src/partnerassessment/components/AssessmentForm.jsx [hunk 1]
```diff
@@ -1,13 +1,55 @@
...
   render() {
+    const {choices, value} = this.props;
     return (
       <div>
-          AssessmentForm: title, textarea, dropdown and save button.
+      <h4>Outreach Effectivness</h4>
+      <span className="pull-right">date</span>
+      <div className="formControlsTextarea">
+      <label>Written Assessment</label><br/>
+        <Textarea value={this.state.value} onChange={e => this.handleTextInput(e)} name="rating-assessment"/><br/>
```
Comment (iartemiev) [inline gulp/src/partnerassessment/components/AssessmentForm.jsx]: It's better to bind that onChange handler to `this` in the constructor and then:
```javascript
onChange={this.handleTextInput}
```

File: gulp/src/partnerassessment/components/AssessmentForm.jsx [hunk 2]
```diff
@@ -1,13 +1,55 @@
...
+      <label>Rate Effectivness</label>
+        <Select
+        choices={choices}
+        name="rating-choices"
+        onChange={e => this.onDropDownChange(e)}
```
Comment (iartemiev) [inline gulp/src/partnerassessment/components/AssessmentForm.jsx]: Same thing here

File: gulp/src/partnerassessment/reducers/assessmentReducer.js [hunk 3]
```diff
@@ -0,0 +1,21 @@
...
+        ...state.assessment,
+        score: rating
+      },
+    };
+  },
+  'UPDATED_ASSESSMENT': (state, action) => {
+    const updated = action.payload;
+    return {
+      ...state,
+    };
```
Comment (iartemiev) [inline gulp/src/partnerassessment/reducers/assessmentReducer.js]: Need to actually set the updated `state.assessment.text` here (this is what is causing the build to fail)

File: gulp/src/partnerassessment/reducers/initialState.js [hunk 4]
```diff
@@ -22,6 +22,32 @@
  */
```
Comment (aljordan82) [inline gulp/src/partnerassessment/reducers/initialState.js]: The rating options could be in its own file.

File: gulp/src/partnerassessment/components/AssessmentForm.jsx [hunk 5]
```diff
@@ -1,13 +1,55 @@
...
+  constructor() {
+  super();
+
+  this.state = {
+    textValue: '',
+    selectValue: '',
+  };
+  this.handleSave = this.handleSave.bind(this);
+}
+onDropDownChange(e) {
```
Comment (aljordan82) [inline gulp/src/partnerassessment/components/AssessmentForm.jsx]: Can we combine all form dispatcher into one dispatch? I think the code will be less cluttered if we have less dispatchers.

Comment (iartemiev) [inline gulp/src/partnerassessment/components/AssessmentForm.jsx]: game-of-thrones-stannis-fewer

Comment (iartemiev) [inline gulp/src/partnerassessment/components/AssessmentForm.jsx]: I'm sorry. I couldn't resist

File: gulp/src/partnerassessment/components/AssessmentForm.jsx [hunk 6]
```diff
@@ -1,13 +1,55 @@
...
+}
+
   render() {
+    const {choices, value} = this.props;
     return (
       <div>
-          AssessmentForm: title, textarea, dropdown and save button.
+      <h4>Outreach Effectivness</h4>
+      <span className="pull-right">date</span>
+      <div className="formControlsTextarea">
```
Comment (aljordan82) [inline gulp/src/partnerassessment/components/AssessmentForm.jsx]: it is better to use kebab-case (dashed)

File: gulp/src/partnerassessment/components/PartnerAssessmentApp.jsx [hunk 7]
```diff
@@ -11,13 +11,14 @@ import RecordContainer from './Records/RecordContainer';
 /* PartnerAssessmentApp */
 class PartnerAssessmentApp extends Component {
   render() {
-    const {loading} = this.props;
+    const {value, choices, dispatch} = this.props;
+              // {loading && <LoadingSpinner />}
```
Comment (aljordan82) [inline gulp/src/partnerassessment/components/PartnerAssessmentApp.jsx]: LoadingSpinner is needed

File: gulp/src/partnerassessment/components/AssessmentForm.jsx (hunk 5 above)
Comment (strempe) [inline gulp/src/partnerassessment/components/AssessmentForm.jsx]: nice!

File: gulp/src/partnerassessment/components/PartnerAssessmentApp.jsx [hunk 8]
```diff
@@ -2,7 +2,7 @@ import React, {Component} from 'react';
 import PropTypes from 'prop-types';
 import {connect} from 'react-redux';
 
-import LoadingSpinner from 'common/ui/LoadingSpinner';
+// import LoadingSpinner from 'common/ui/LoadingSpinner';
```
Comment (aljordan82) [inline gulp/src/partnerassessment/components/PartnerAssessmentApp.jsx]: Spinner is needed

File: gulp/src/partnerassessment/api.js [hunk 9]
```diff
@@ -3,4 +3,7 @@ export default class Api {
   constructor(myJobsApi) {
     this.myJobsApi = myJobsApi;
   }
+  async updatedAssessment(information) {
+    return await this.api.post('/prm/api/assessments?partner_id=<id>' + information);
```
Comment (aljordan82) [inline gulp/src/partnerassessment/api.js]: what is <id> ?

**Comment by koonsies** [comment]
Something funny going on here.. Has part of your code been merged already? Seeing a very small diff.

**Pull request closed without merging**